### ISSUE TITLE
Android - General - Poor Color Contrast of Step Indicator (Sub-Navigation)

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#10]
The color contrast of the “Steps” indicator at the top of the screen does not have enough color contrast and needs to be 4.5 or higher to pass. The light gray numbered circles (#D3D3D3 ), and the lighter gray background (#EFEFEF ) has a color contrast ratio of 1.3:1 and needs to be 4.5:1 or higher to pass.

There needs to be enough color contrast between the circles and the background along with the circles and the number and the circles and the lines connecting them unless those use the same/similar color to make it look connected. Plus, if they are supposed to be used to show which Step a person is on then that need to be considered too.

Please see screenshot.

![android - bookmarks screen - poor color contrast](https://user-images.githubusercontent.com/10423487/57105828-df44df00-6cf9-11e9-8b18-1e475fb7d1c4.png)

**Comment by KyleLivingstonDOL** [comment]
Thanks John. We will address this.

**Comment by KyleLivingstonDOL** [comment]
Hi John,

We've updated this and will be demo'ed on the next 508 compliance check after Sprint 7.

Best,
Kyle

**Comment by KyleLivingstonDOL** [comment]
> Hi John,
> 
> We've updated this and will be demo'ed on the next 508 compliance check after Sprint 7.
> 
> Best,
> Kyle

UAT for Sprint 7 (which is ongoing right now) will start on June 10.

**Comment by jfc3-dol** [comment]
There is not enough color contrast between the gold half circle ( #FDB81E ) and the gray background ( #EFEFEF ). The color contrast ratio is 1.52:1 and needs to be 4.5:1 or higher to pass. It's improved but not enough.

The color contrast ratio between the gold half circle ( #FDB81E ) and the blue number and connector line ( #0071BC ) is 2.95:1.

**Comment by KyleLivingstonDOL** [comment]
Thanks John,

As we are trying to match 3 colors (the blue numbering, the white interior background, and the gray shading around the timeline), we cannot seem to find a color that denotes what the timeline bar is supposed to indicate: that a user is on a current step and it is in progress. As such, we would like to put this issue on hold as we would need to go back to the beginning of app development to redo the coloring and color palate chosen.

Thanks,
Kyle

**Comment by jfc3-dol** [comment]
Before you give up, a quick thought would be instead of using the gold half circle why not put a blue circle round the blue number and have the background white and darken the gray a slightest bit which will then pass using the blue you are using already.

**Comment by KyleLivingstonDOL** [comment]
> Before you give up, a quick thought would be instead of using the gold half circle why not put a blue circle round the blue number and have the background white and darken the gray a slightest bit which will then pass using the blue you are using already.

Hi John,

This is a good idea except we also have to differentiate from the blue progress bar as well, correct? If so, then the blue ring around the blue number wouldn't have enough contrast from the blue progress bar.

-Kyle

**Comment by jfc3-dol** [comment]
Do you need the progress bar? When I went to look at Step 4, I noticed the progress bar between Steps 1 - 4 when I might have only looked at Step 1 and Step 4. Is that done systematically?

If the current Step is highlighted with a blue outer circle border with a blue number that matched or connected to the progress bar as long as they are the same color would be fine. Or if you wanted to make it a fully colored blue circle that matched the progress bar with white text for the number that would be good too to denote the current Step.

My thought was a blue border around the circle that matched the progress bar and had the same colored number.

**Comment by KyleLivingstonDOL** [comment]
The progress bar is something that we built into the app utilizing our users' feedback. If we were to change it, we'd need to redesign the progress "bar," which we currently do not have time for at this point. We would like to keep the blue progress bar as that is what our users are expecting.

**Comment by jfc3-dol** [comment]
That's fine if you keep it. To make it accessible the colors need to have enough color contrast.

You don't think you could remove the half gold circle and and do something like the attached? Where the progress bar stays but the current Step has a circle around it that matches in color to the of the progress bar with the same gaps and gray background.

 I had to use Microsoft Paint so it's not pretty and the colors are what I had available to me..

![comply chain example sub navigation using a circle to denote current step](https://user-images.githubusercontent.com/10423487/59538654-85befb00-8ec8-11e9-9d5c-33c810b7e0b2.png)

**Comment by KyleLivingstonDOL** [comment]
Thanks for this John; we appreciate your insight into this. I think at this point, we want to keep the current timeline design. I think it makes the most sense for how we want it to function. We want to provide our users a visual understanding of where they are in Comply Chain. I think we can revisit this in another release as we brainstorm other ways to make the timeline more accessible.

**Comment by jfc3-dol** [comment]
If that is the case and updates won't be made now, I will mark this as for next release as long as you know this and other issues marked as "Future Release" could be a problem for people, and they could complain that the color contrast, in this case, is an issue.

**Comment by jfc3-dol** [comment]
The English version shows the breadcrumbs are better than they previously were but still need improvement in the next version.

Please see the attached screenshot.

![comply chain - english - breadcrumbs are better than previously but still need improvement in the next version](https://user-images.githubusercontent.com/10423487/64364162-5360fd80-cfe0-11e9-8f2a-f6b43ae1393b.png)

**Comment by jfc3-dol** [comment]
The same color contrast issue is in the application as previously. Also, the four-show/hide buttons on the Step screens have poor color contrast, and if there is no issue already, I will open a new one for it.

Please see the attached screenshot.

![ilab comply chain - step progress bar color contrast needs improving](https://user-images.githubusercontent.com/10423487/116712565-f9c1f980-a9a1-11eb-9613-f2db06c595bf.jpg)

**Comment by jfc3-dol** [comment]
@pbhatt17 

The color contrast for the Step indicator (sub-navigation) is still an issue that needs to be figured out.

**Comment by MikePulsiferDOL** [comment]
Testing latest version has contrast ratio of 7.1:1

<img width="350" alt="Screen Shot 2022-01-18 at 2 13 53 PM" src="https://user-images.githubusercontent.com/1970829/150003342-e2667cdc-76d3-4d9e-abe3-5abf73f2fcc5.png">
